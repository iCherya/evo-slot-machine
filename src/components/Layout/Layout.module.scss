@import '@/styles/variables.scss';

.wrapper {
  max-width: $breakpoint-xxl;
  padding: $spacing-base;
  display: flex;
  flex-direction: column;
  gap: $spacing-large;
  margin: 0 auto;
}
