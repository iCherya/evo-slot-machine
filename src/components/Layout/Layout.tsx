import React from 'react';

import { Header } from '@/components/Header';
import { Footer } from '@/components/Footer';

import { IProviderProps } from '@/types';

import styles from './Layout.module.scss';

export const Layout: React.FC<IProviderProps> = ({ children }) => {
  return (
    <div className={styles.wrapper}>
      <Header />
      <main className={styles.childrenWrapper}>{children}</main>
      <Footer />
    </div>
  );
};
