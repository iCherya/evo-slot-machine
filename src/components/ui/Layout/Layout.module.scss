@import '@/styles/variables.scss';

.wrapper {
  max-width: $breakpoint-lg;
  padding: $spacing-base;
  display: flex;
  flex-direction: column;
  margin: 0 auto;
}

.childrenWrapper {
  background-color: $color-base-light;
  color: $color-base-dark;
  border-radius: $spacing-base;
  margin-bottom: $spacing-small;
  border-width: 0;
}
