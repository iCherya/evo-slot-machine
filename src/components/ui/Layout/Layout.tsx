import React from 'react';

import { Header } from '@/components/ui/Header';
import { Footer } from '@/components/ui/Footer';

import { IProviderProps } from '@/types';

import styles from './Layout.module.scss';

export const Layout: React.FC<IProviderProps> = ({ children }) => {
  return (
    <div className={styles.wrapper}>
      <Header />
      <main className={styles.childrenWrapper}>{children}</main>
      <Footer />
    </div>
  );
};
