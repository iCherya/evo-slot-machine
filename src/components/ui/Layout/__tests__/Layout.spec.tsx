import { render } from '@testing-library/react';

import { Layout } from '@/components/ui/Layout';

jest.mock('@/components/ui/Header', () => ({
  Header: () => <div data-testid="header" />,
}));

jest.mock('@/components/ui/Footer', () => ({
  Footer: () => <div data-testid="footer" />,
}));

jest.mock('@/components/ui/Settings', () => ({
  Settings: () => <div data-testid="settings" />,
}));

describe('Layout', () => {
  const Child = () => <div data-testid="child">Child component</div>;

  it('should render Header, Footer and Settings components', () => {
    const { container, getAllByTestId } = render(
      <Layout>
        <Child />
      </Layout>,
    );

    expect(getAllByTestId('header').length).toBe(1);
    expect(getAllByTestId('footer').length).toBe(1);
    expect(getAllByTestId('settings').length).toBe(1);

    expect(container).toMatchSnapshot();
  });

  it('should render children components', () => {
    const { container, getAllByTestId } = render(
      <Layout>
        <Child />
        <Child />
        <Child />
      </Layout>,
    );

    expect(getAllByTestId('child').length).toBe(3);

    expect(container).toMatchSnapshot();
  });
});
