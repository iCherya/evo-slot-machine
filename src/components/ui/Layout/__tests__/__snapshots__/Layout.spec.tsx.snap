// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render Header, Footer and Settings components 1`] = `
<div>
  <div>
    <div
      data-testid="header"
    />
    <main>
      <div
        data-testid="child"
      >
        Child component
      </div>
    </main>
    <div
      data-testid="footer"
    />
    <div
      data-testid="settings"
    />
  </div>
</div>
`;

exports[`Layout should render children components 1`] = `
<div>
  <div>
    <div
      data-testid="header"
    />
    <main>
      <div
        data-testid="child"
      >
        Child component
      </div>
      <div
        data-testid="child"
      >
        Child component
      </div>
      <div
        data-testid="child"
      >
        Child component
      </div>
    </main>
    <div
      data-testid="footer"
    />
    <div
      data-testid="settings"
    />
  </div>
</div>
`;
